List Lab/src/NikitaS/MyLinkedList.java

```diff
@@ -0,0 +1,108 @@
+package NikitaS;
+
+import java.util.Iterator;
+
+public class MyLinkedList<T> implements Iterable<T>{
+
+	private Node<T> head;
+	private int size = 0;
+
+	public int getSize() {
+		return size;
+	}
+
+	public boolean remove(int index) {
+		//Set both initial nodes to be head.
+		if (size ==0){
+			return false;
+		}
+		size--;
+		Node<T> nextNode= head;
+		Node<T> previousNode=head; //To avoid null exceptions
+		for (int i=0;i<index;i++) { //Use "<" because the 0th index is not the head, but the node the head is pointing to.
+			previousNode= nextNode;//Keep track of previous node. On the first iteration it will remain head.
+			nextNode = nextNode.getNext(); //Get current node's reference to next node.
+		}
+		Node<T> newTarget = nextNode.getNext(); //Get the nextNode reference of the node to be removed.
+		previousNode.setNext(newTarget); //Set that new reference to the node that came before the previous one.
```

Comment:
This works great for `0 < index < size`, but what if `index` is zero, i.e. the caller wants to remove the head? I don't think your algorithm will change anything in that case.